docker/archive/src.go

```diff
@@ -34,3 +35,83 @@ func newImageSource(ctx context.Context, sys *types.SystemContext, ref archiveRe
 func (s *archiveImageSource) Reference() types.ImageReference {
 	return s.ref
 }
+
+// MultiImageSourceItem is a reference to _one_ image in a multi-image archive.
+// Note that MultiImageSourceItem implements types.ImageReference.  It's a
+// long-lived object that can only be closed via it's parent MultiImageSource.
+type MultiImageSourceItem struct {
+	*archiveReference
+	tarSource *tarfile.Source
+}
+
+// Manifest returns the tarfile.ManifestItem.
+func (m *MultiImageSourceItem) Manifest() (*tarfile.ManifestItem, error) {
```

Comment:
Nothing but `RepoTags` is _really_ usable by callers, see e.g. the use of `Config` in https://github.com/containers/podman/pull/6811 ; so I’m a bit reluctant to commit to this as an API. OTOH, we clearly _can_ keep this stable.